### PR TITLE
all SequenceIterators implement parse() but it was not explicit

### PR DESCRIPTION
**Pull request by mr-c**
- [x] I hereby agree to dual licence this and any previous contributions under both
the _Biopython License Agreement_ **AND** the _BSD 3-Clause License_.

- [x] I have read the ``CONTRIBUTING.rst`` file, have run ``flake8`` locally, and
understand that AppVeyor and TravisCI will be used to confirm the Biopython unit
tests and style checks pass with these changes.

- [ ] I have added my name to the alphabetical contributors listings in the files
``NEWS.rst`` and ``CONTRIB.rst`` as part of this pull request, am listed
already, or do not wish to be listed. (*This acknowledgement is optional.*)

### CODE REVIEW

File: Bio/SeqIO/Interfaces.py [hunk 1]
```diff
@@ -91,6 +94,10 @@ def __iter__(self):
         """
         return self
 
+    @abstractmethod
+    def parse(self, handle: IO[Any]) -> Iterator[SeqRecord]:
```
Comment (peterjc) [inline Bio/SeqIO/Interfaces.py:98]: Can you take out the annotation for now please.

Comment (mr-c) [reply Bio/SeqIO/Interfaces.py]: Done

**Comment by chris-rands** [comment]
@mr-c Given your contributions here, perhaps you would be willing to review  #2732 and #2731?

**Comment by mr-c** [comment]
@peterjc Can this be merged?

**Comment by peterjc** [comment]
I was hoping for comment from @mdehoon as he refactored this code last.

**Pull request merged**